packages/admin/src/api.ts

```diff
@@ -66,10 +74,52 @@ export class AdminClient extends BaseApiClient {
     });
   }
 
-  public async createProposal(proposal: CreateProposalRequest): Promise<ProposalResponseWithUrl> {
+  // added separate from CreateProposalRequest type as the `simulate` boolean is contained within defender-client
+  public async createProposal(
+    proposal: CreateProposalRequest & { simulate?: boolean; overrideSimulationOpts?: SimulationTransaction },
+  ): Promise<ProposalResponseWithUrl> {
     return this.apiCall(async (api) => {
       const response = (await api.post('/proposals', proposal)) as ProposalResponse;
-      return { ...response, url: getProposalUrl(response) };
+
+      // handle simulation
+      let simulation: SimulationResponse | undefined = undefined;
+      const isBatchProposal = (contract: PartialContract | PartialContract[]): contract is PartialContract[] =>
+        isArray(contract);
+      if (proposal.simulate) {
+        const overrideData = proposal.overrideSimulationOpts?.transactionData.data;
+        // data property is overriden in backend for batch proposals
+        let data = overrideData ?? 'batch_override';
+        if (!isBatchProposal(proposal.contract) && !overrideData) {
+          if (!proposal.contract.abi) {
+            // no ABI found, request user to pass in `data` in overrideSimulationOpts
+            throw new Error(
+              'Simulation requested without providing ABI. Please provide the contract ABI or use the `overrideSimulationOpts` to provide the data property directly.',
+            );
+          }
+          const contractInterface = new Interface(proposal.contract.abi);
+
+          // this is defensive and should never happen since createProposal schema validation will fail without this property defined.
+          if (!proposal.functionInterface) {
+            // no function selected, request user to pass in `data` in overrideSimulationOpts
+            throw new Error(
+              'Simulation requested without providing function interface. Please provide the function interface or use the `overrideSimulationOpts` to provide the data property directly.',
+            );
+          }
+          data = contractInterface.encodeFunctionData(proposal.functionInterface.name!, proposal.functionInputs);
+        }
+        simulation = await this.simulateProposal(response.contractId, response.proposalId, {
```

Comment:
what happens if proposal creation was successful but simulation fails (due to timeout for instance)?

Reply:
The new commit introduces some checks that ensures the simulation doesn't interrupt createProposal unnecessarily.